pillar/nginx/mitxpro_redirect.sls

```diff
@@ -0,0 +1,39 @@
+nginx:
+  servers:
+    managed:
+      mitxpro-redirect:
+        enabled: True
+        config:
+          - server:
+            - server_name: mitxpro.mit.edu
+            - listen:
+                - 80
+            - listen:
+                - 443
+                - ssl
+            - listen:
+              - '[::]:80'
+              - '[::]:443'
+              - ssl
+              # The certificate uses subject alternative names, including mitxpro.mit.edu.
+              - ssl_certificate: /etc/letsencrypt/live/amps-web.amps.ms.mit.edu/cert.pem
```

Comment:
Should this be set to a cert for the mtixpro domain?

Reply:
I think that I planned it this way so that the certificate could be acquired through Letsencrypt prior to the DNS change. The domain `mitxpro.mit.edu` will still point to the edX servers when we first request the certificate. If we make the request for `amps-web.amps.ms.mit.edu` with a SAN of `mitxpo.mit.edu` I expect it to succeed because `amps-web.amps.ms.mit.edu` resolves correctly.

I'm not sure how much the downtime matters. If we get the certificate for `mitxpro.mit.edu` then there will be some time between when DNS cuts over and then we run the `certbot` script when someone could get an SSL error.

Reply:
It's been a while since I worked on this last, and another thing that I admit I can't figure out now is why I added the `letsencrypt` state to `salt/top.sls` but there's no pillar data to really configure the certificate. It could be that I just got distracted and never finished that work.